jaggery-modules/carbon-metrics/module/scripts/timers/timers.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function(metrics) {
+    var MetricManager = org.wso2.carbon.metrics.manager.MetricManager;
+    var Level = org.wso2.carbon.metrics.manager.Level;
+    var pushContext = function(timer) {
+        if (typeof WSO2_METRICS_TIMERS === 'undefined') {
+            WSO2_METRICS_TIMERS = [];
+        }
+        WSO2_METRICS_TIMERS.push(timer);
+    };
+    var popContext = function() {
+        if (typeof WSO2_METRICS_TIMERS === 'undefined') {
+            return null; //Do nothing since there are no timers
+        }
+        return WSO2_METRICS_TIMERS.pop();
+    };
+    var getlastInvocation = function() {
+        if (typeof(WSO2_METRICS_LAST_INVOCATION) === 'undefined') {
```

Comment:
Hi @thushara35 ,Thank you for the suggestion,
I have tried the above simplified if statement and it seems that expression is going to fail if the variable used in the if statement (in this case  **WSO2_METRICS_LAST_INVOCATION**) haven't even declared at the moment this expression get executed.

For example:

```javascript
var abc;
if (!a) {
    console.log("Not declared!") // This statement get executed
}
```

but if the variable hasn't even declared then the above statement will throw 

> ReferenceError: **abc** is not defined

and also it is recommended[(1)](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/undefined#Strict_equality_and_undefined) use strict equality[(2)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#Strict_equality_using) operator rather than the standard equality operator when checking for an undefined state of a variable.

(1) : https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/undefined#Strict_equality_and_undefined

(2) : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#Strict_equality_using